File: leetcode/0442-find-all-duplicates-in-an-array/solution.py
from typing import List


class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        result = []

        for x in nums:
            idx = abs(x) - 1
            if nums[idx] < 0:
                result.append(idx + 1)
            else:
                nums[idx] = -nums[idx]

        return result

File: leetcode/0442-find-all-duplicates-in-an-array/solution.cpp
#include <vector>
#include <cstdlib>
using namespace std;

class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> result;

        for (int x : nums) {
            int idx = abs(x) - 1;
            if (nums[idx] < 0) {
                result.push_back(idx + 1);
            } else {
                nums[idx] = -nums[idx];
            }
        }

        return result;
    }
};
